src/modules/grubcfg/main.py

```diff
@@ -214,6 +217,10 @@ def modify_grub_default(partitions, root_mount_point, distributor):
         kernel_params.append(f"rd.luks.uuid={swap_outer_uuid}")
     if swap_outer_mappername:
         kernel_params.append(f"resume=/dev/mapper/{swap_outer_mappername}")
+        
+    # When using rd.luks.name you can omit rd.luks.uuid
+    if have_systemd:
+        kernel_params.append(f"rd.luks.name={partition['luksUuid']}={partition['luksMapperName']}")
```

Comment:
This is going to add this to the existing options even if the partition isn't encrypted.

Also, if the partition is encrypted, it will already have something else added to the options for encryption.

Isn't the logic when using the sd-encrypt the same as when dracut is applied above?  Could we not just use that code?

Reply:
The mkinitcpio_systemd variable works only if there is sd-encrypt in the hooks, sd-encrypt in turn is added only if unencrypted_separate_boot works in initcpiocfg, that is, in theory, if the partition is not encrypted, it just does not add anything.

Reply:
What's supposed to happen? sd-encrypt will not be shipped with an unencrypted partition in any case, so there will be no errors either.

Reply:
Or do you want to just take the code from dracut and change it somehow ?

```sh
    if have_systemd or have_dracut:
        for partition in partitions:
            if partition["fs"] == "linuxswap" and not partition.get("claimed", None):
                # Skip foreign swap
                continue
            has_luks = "luksMapperName" in partition
            if partition["fs"] == "linuxswap" and not has_luks:
                swap_uuid = partition["uuid"]

            if partition["fs"] == "linuxswap" and has_luks:
                swap_outer_uuid = partition["luksUuid"]
                swap_outer_mappername = partition["luksMapperName"]

            if partition["mountPoint"] == "/" and has_luks:
                cryptdevice_params = [f"rd.luks.uuid={partition['luksUuid']}"]
    else:
        for partition in partitions:
            if partition["fs"] == "linuxswap" and not partition.get("claimed", None):
                # Skip foreign swap
                continue
            has_luks = "luksMapperName" in partition
            if partition["fs"] == "linuxswap" and not has_luks:
                swap_uuid = partition["uuid"]

            if partition["fs"] == "linuxswap" and has_luks:
                swap_outer_mappername = partition["luksMapperName"]

            if partition["mountPoint"] == "/" and has_luks:
                cryptdevice_params = [
                    f"cryptdevice=UUID={partition['luksUuid']}:{partition['luksMapperName']}",
                    f"root=/dev/mapper/{partition['luksMapperName']}"
                ]
```

Reply:
Right, I am asking if the dracut code would work "as-is" when using the sd-encrypt hook.  Assuming we made the code follow that path when the sd-encrypt hook was enabled.

Also, we shouldn't call it `has_systemd`, that would be a very confusing name.  A distro could have systemd and not be using that hook or even mkinitcpio.

Reply:
I guess if I do as in the snippet above, everything should work correctly

Reply:
I was originally pointed out the need to redo grubcfg by @bkmo, maybe he can tell me how else to implement it.

Reply:
There's a line in the loader
``sh
if have_dracut:
                cryptdevice_params = [f "rd.luks.uuid={partition['luksUuid']}"]]
```
I guess for systemd it is enough to add this by analogy
```sh
if uses_sd-encrypt:
                cryptdevice_params = [f "rd.luks.name={partition['luksUuid']}={partition['luksMapperName']}"]]
```

Reply:
You shouldn't need to.  It shouldn't have to be `rd.luks.name=`, `rd.luks.uuid=` should also work.

Reply:
So you can combine two conditions into one, just like in grubcfg

Reply:
I seem to have added all the edits that were required of me now waiting for edits on initcpiocfg if they are available

Reply:
does the bootldr module also need an addition for systemd added to the dracut section?